src/graphql/data-sources/serlo.ts

```diff
@@ -248,6 +248,27 @@ export class SerloDataSource extends CacheableDataSource {
     return this.cacheAwareGet({ path: `/api/threads/${id}` })
   }
 
+  public async createThread(
+    payload: MutationCreateThreadArgs
+  ): Promise<ThreadsPayload> {
+    const firstCommentId = await this.customPost<number>({
+      path: `/api/create-thread/`,
+      body: payload,
+    })
+    const threads = await this.getThreadIds({ id: payload.object })
+    threads.firstCommentIds.push(firstCommentId)
+    return await this.setCache({
+      key: this.getCacheKey(`/api/threads/${payload.object}`),
+      update: () =>
+        this.customPost<ThreadsPayload>({
+          path: `/api/threads/${payload.object}`,
+          body: {
+            firstCommentIds: [threads],
+          },
+        }),
```

Comment:
```suggestion
    return await toThreadPayload(dataSource.serlo, firstCommentId)
```

This should be part of the resolver -> see comment above, Note that you have caching in the methods `toThreadPayload` uses and thus you do not need to add them here.